pkg/plugins/document/firestore/firestore.go

```diff
@@ -83,13 +83,53 @@ func (s *FirestoreDocService) Delete(key *sdk.Key) error {
 
 	doc := s.getDocRef(key)
 
+	// 	Delete any sub collection documents
+	iter := doc.Collections(s.context)
+	for {
+		subColRef, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("error deleting value: %v", err)
+		}
+
+		for {
+			iter := subColRef.Limit(100).Documents(s.context)
+			numDeleted := 0
+
+			batch := s.client.Batch()
+			for {
+				subDoc, err := iter.Next()
```

Comment:
These lines could probably be merged into the 'for' loop for clarity on what's being looped over

```go
for subDoc, err := iter.Next(); err != iterator.Done {
```

Reply:
Not sure the alternative is much better:
```
subDoc, err := docsIterator.Next()
for ; err != iterator.Done; subDoc, err = docsIterator.Next() {
    if err != nil {
        return err
    }

    batch.Delete(subDoc.Ref)
    numDeleted++
}
```

Reply:
you could include init in the for loop:

```go
for subDoc, err := docsIterator.Next(); err != iterator.Done; subDoc, err = docsIterator.Next() {
    if err != nil {
        return err
    }

    batch.Delete(subDoc.Ref)
    numDeleted++
}
```

Reply:
Done